monkey/monkey_island/cc/ui/src/components/report-components/attack/SelectedTechnique.js

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import Collapse from '@kunukn/react-collapse';
+
+import AttackReport from '../AttackReport';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import {faQuestionCircle} from '@fortawesome/free-solid-svg-icons';
+
+const classNames = require('classnames');
+
+class SelectedTechnique extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      techniques: this.props.techniques,
+      techComponents: this.props.techComponents,
+      selectedTechnique: this.props.selected
+    };
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.selected !== prevProps.selected || this.props.techniques !== prevProps.techniques) {
+     this.setState({ selectedTechnique: this.props.selected,
+                     techniques: this.props.techniques})
+    }
+  }
+
+  getSelectedTechniqueComponent(tech_id) {
+    const TechniqueComponent = this.state.techComponents[tech_id];
+    return (
+      <div key={tech_id} className={classNames('collapse-item', {'item--active': true})}>
+        <button className={classNames('btn-collapse',
+                                      'selected-technique',
+                                      AttackReport.getComponentClass(tech_id, this.state.techniques))}>
+          <span>{this.state.techniques[tech_id].title}</span>
+          <span>
+            <a href={this.state.techniques[tech_id].link} target='_blank' className={'link-to-technique'}>
+              <FontAwesomeIcon icon={faQuestionCircle}/>
+            </a>
+          </span>
+        </button>
+        <Collapse
+          className='collapse-comp'
+          isOpen={true}
+          render={() => {
+            return (<div className={`content ${tech_id}`}>
+              <TechniqueComponent data={this.state.techniques[tech_id]}/>
+            </div>)
+          }}/>
+      </div>
```

Comment:
Can we refactor this into a component and reuse it in the TechniquesDropdown big list?

Reply:
Collapse is already a component and button is a bit different ( doesn't have chevron icon)